src/main/java/com/metamx/cache/CaffeineCache.java

```diff
@@ -45,15 +49,31 @@
 
   public static CaffeineCache create(final CaffeineCacheConfig config)
   {
-    Caffeine<Object, Object> builder = Caffeine.newBuilder().recordStats();
+    return create(config, null);
+  }
+
+  // Used in testing
+  public static CaffeineCache create(final CaffeineCacheConfig config, @Nullable final Executor executor)
+  {
+    Caffeine builder = Caffeine.newBuilder().recordStats();
     if (config.getExpiration() >= 0) {
       builder = builder
           .expireAfterAccess(config.getExpiration(), TimeUnit.MILLISECONDS);
     }
     if (config.getMaxSize() >= 0) {
       builder = builder
-          .maximumSize(config.getMaxSize())
-          .maximumWeight(config.getMaxSize());
+          .maximumWeight(config.getMaxSize())
+          .weigher(new Weigher<NamedKey, byte[]>()
```

Comment:
Yeah, the inference is done during the build to capture the types
incrementally. Worked great in Java 5, but lamdas infer Object. By
specifying the type on the lambda it works nicely again, so casting isn't
required.

On Thu, Mar 31, 2016 at 12:36 PM, Charles Allen <notifications@github.com>
wrote:

> In src/main/java/com/metamx/cache/CaffeineCache.java
> <https://github.com/metamx/druid-cache-caffeine/pull/7#discussion_r58114689>
> :
>
> >      if (config.getExpiration() >= 0) {
> >        builder = builder
> >            .expireAfterAccess(config.getExpiration(), TimeUnit.MILLISECONDS);
> >      }
> >      if (config.getMaxSize() >= 0) {
> >        builder = builder
> > -          .maximumSize(config.getMaxSize())
> > -          .maximumWeight(config.getMaxSize());
> > +          .maximumWeight(config.getMaxSize())
> > +          .weigher(new Weigher<NamedKey, byte[]>()
>
> It was due to the builder being <Object, Object> I simply forced the
> typing when creating the builder and things are happy now.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/metamx/druid-cache-caffeine/pull/7/files/00b3f0a3a5dec232e12bc135a6ccc27a738bc9b7#r58114689>
>